frontend/packages/console-app/src/components/pdb/PDBForm.tsx

```diff
@@ -0,0 +1,364 @@
+import * as React from 'react';
+import {
+  ActionGroup,
+  Button,
+  Form,
+  TextInput,
+  FormGroup,
+  GridItem,
+  Grid,
+  Text,
+  Title,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import i18next from 'i18next';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { MatchLabels } from '@console/dynamic-plugin-sdk/src/api/common-types';
+import {
+  ButtonBar,
+  history,
+  resourcePathFromModel,
+  SelectorInput,
+} from '@console/internal/components/utils';
+import { FieldLevelHelp } from '@console/internal/components/utils/field-level-help';
+import { PodDisruptionBudgetModel } from '@console/internal/models';
+import { k8sCreate, k8sPatch, PodDisruptionBudgetKind, Patch } from '@console/internal/module/k8s';
+import AvailabilityRequirementPopover from './AvailabilityRequirementPopover';
+import { pdbToK8sResource, FormValues, intOrPercentage } from './pdb-models';
+
+const formValuesFromK8sResource = (from: PodDisruptionBudgetKind): FormValues => {
```

Comment:
Actually this was in the `models.ts` in my initial PR but was advised to place it here.

Reply:
Who advised that? I'd definitely like to overrule that. Co-locating code together because it's only used in one spot is not useful to making a proper structured application. Clean readable files, with little noise is how you maintain such a large codebase.